tutorial/server/axum/src/startup.rs

```diff
@@ -0,0 +1,51 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use webauthn_rs::prelude::*;
+
+// 2. Configure the Webauthn instance by using the WebauthnBuilder. This defines
+// the options needed for your site, and has some implications. One of these is that
+// you can NOT change your rp_id (relying party id), without invalidating all
+// webauthn credentials. Remember, rp_id is derived from your URL origin, meaning
+// that it is your effective domain name.
+
+pub struct Data {
+    pub name_to_id: HashMap<String, Uuid>,
+    pub keys: HashMap<Uuid, Vec<PassKey>>,
+}
+
+#[derive(Clone)]
+pub struct AppState {
+    // Webauthn has no mutable inner state, so Arc and read only is sufficent.
+    // Alternately, you could use a reference here provided you can work out
+    // lifetimes.
+    pub webauthn: Arc<Webauthn>,
+    // This needs mutability, so does require a mutex.
+    pub users: Arc<Mutex<Data>>,
+}
+
+impl AppState {
+    pub fn new() -> Self {
+        // Effective domain name.
+        let rp_id = "localhost";
+        // Url containing the effective domain name
+        // MUST include the port number!
+        let rp_origin = Url::parse("http://localhost:8080").expect("Invalid URL");
+        let builder = WebauthnBuilder::new(rp_id, &rp_origin).expect("Invalid configuration");
+
+        // Now, with the builder you can define other options.
+        // Set a "nice" relying party name. Has no security properties and
+        // may be changed in the future.
+        let builder = builder.rp_name("Panda5");
```

Comment:
Should the rp_name here be "Webauthn-RS Axum Demo"?

Reply:
Yes it should